### PR TITLE
Warning fix

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will update the changelog and release version date

### Proposed changes

- update changelog
- update release related docs
- fix some warnings
- update units library for warning fix

### CODE REVIEW

File: src/helics/shared_api_library/MessageFederate.h [hunk 1]
```diff
@@ -277,7 +277,7 @@ HELICS_EXPORT helics_message_object helicsEndpointGetMessageObject(helics_endpoi
  *
  * @details The message is empty and isValid will return false since there is no data associated with the message yet.
  *
- * @param fed the endpoint object to associate the message with
+ * @param endpoint The endpoint object to associate the message with
```
Comment (nightlark) [inline src/helics/shared_api_library/MessageFederate.h:280]:
```suggestion
 * @param endpoint The endpoint object to associate the message with.
```

File: docs/ROADMAP.md [hunk 2]
```diff
@@ -2,30 +2,22 @@
...
...ich should be supported on Ubuntu 18.04 repositories. Minimum Boost version will also be updated though Boost is becoming less critical for the HELICS core so may not be that important. The likely minimum tested target will likely be 1.65.1 though the core might work with older versions and its use can be disabled completely. Certain features may require a newer boost version(1.70) than what would be available on Ubuntu 18.04. General target requirements will allow HELICS to build on the most recent 2 LTS versions of Ubuntu using readily available repo packages. Minimum required compilers for building on macOS and systems using ICC will include Xcode 10 and ICC 19. The minimum ZMQ version will also be bumped up to 4.2. General policy for Mac builds will be supporting Xcode compilers on versions of MacOS that receive security upgrades which is generally the last 3 versions, though 10.1+ and 11 will likely be the only 2 supported at HELICS 3.0 release, and support minor releases for at least 2 years.   MSVC compilers will be supported for at least 2 years from release date, an appropriate CMake (meaning slightly newer than the compiler) will also be required for Visual studio builds.
```
Comment (nightlark) [inline docs/ROADMAP.md]:
```suggestion
Upgrade minimum compilers and build systems. Currently planned minimum targets are gcc 7.0, clang 5.0, MSVC 2017 15.7, XCode 10.1, CMake 3.10. This is a setup which should be supported on Ubuntu 18.04 repositories. Minimum Boost version will also be updated though Boost is becoming less critical for the HELICS core so may not be that important. The likely minimum tested target will likely be 1.65.1 though the core might work with older versions and its use can be disabled completely. Certain features may require a newer boost version(1.70) than what would be available on Ubuntu 18.04. General target requirements will allow HELICS to build on the most recent 2 LTS versions of Ubuntu using readily available repo packages. Minimum required compilers for building on macOS and systems using ICC will include Xcode 10 and ICC 19. The minimum ZMQ version will also be bumped up to 4.2. General policy for Mac builds will be supporting Xcode compilers on versions of MacOS that receive security upgrades which is generally the last 3 versions, though 10.1+ and 11 will likely be the only 2 supported at HELICS 3.0 release, and support minor releases for at least 2 years.   MSVC compilers will be supported for at least 2 years from release date, an appropriate CMake (meaning slightly newer than the compiler) will also be required for Visual Studio builds.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,10 +44,18 @@ Everything within a major version number should be code compatible (with the exc
 - "version" and "version_all" queries to get the local version string and the version strings of all the cores/brokers in the federation
 - A few missing operations to the C++98 interface for Message objects, add `helicsMessageClone` and `helicsEndpointCreateMessage` functions in the C interface. Add a test case for some of the C++98 message operations.
 - `helicsQuerySetTarget` and `helicsQuerySetQueryString` operations to modify an existing query in the C interface
-
-### Deprecated
-
-### Removed
+- a set of reduction operations for multi-input handling on inputs,  options for setting input source priority and the number of expected connections.
```
Comment (nightlark) [inline CHANGELOG.md]:
```suggestion
- A set of reduction operations for multi-input handling on inputs,  options for setting input source priority and the number of expected connections
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,16 +8,34 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+- Update units library to include some typical natural gas units and conversions
+- Use a separate action for automated pr generation
+- Update the CLI11 library
+- The setOption/getOption functions now take an int32_t as a value instead of a boolean.  This does not change the API since in the C library the `helics_bool` was already an int.  
+- In the case of multiple sources, `getInjectionType`, and `getInjectionUnits` now will return a json string vector.
+- The CMake build generation now uses a central location for all build artifacts instead of individual directories.  
+- Updated the ASIO library to 1-16
+- Minor updates to the clang-format to allow better alignment and comment reflow
+- Numerous code refactorings to simplify and clean up code
+- Move all ZMQ related items to the network library
```
Comment (nightlark) [inline CHANGELOG.md:26]:
```suggestion
- Move all ZMQ related items to the network library
- Updated Python packages DLL load failed error to suggest installing the latest Visual C++ Runtime
```

File: docs/ROADMAP.md [hunk 5]
```diff
@@ -2,30 +2,22 @@
...
...which should be supported on Ubuntu 18.04 repositories. Minimum Boost version will also be updated though Boost is becoming less critical for the HELICS core so may not be that important. The likely minimum tested target will likely be 1.65.1 though the core might work with older versions and its use can be disabled completely. Certain features may require a newer boost version(1.70) than what would be available on Ubuntu 18.04. General target requirements will allow HELICS to build on the most recent 2 LTS versions of Ubuntu using readily available repo packages. Minimum required compilers for building on macOS and systems using ICC will include Xcode 10 and ICC 19. The minimum ZMQ version will also be bumped up to 4.2. General policy for Mac builds will be supporting Xcode compilers on versions of MacOS that receive security upgrades which is generally the last 3 versions, though 10.1+ and 11 will likely be the only 2 supported at HELICS 3.0 release, and support minor releases for at least 2 years. MSVC compilers will be supported for at least 2 years from release date, an appropriate CMake (meaning slightly newer than the compiler) will also be required for Visual studio builds.
```
Comment (nightlark) [inline docs/ROADMAP.md:20]:
```suggestion
Upgrade minimum compilers and build systems. Currently planned minimum targets are gcc 7.0, clang 5.0, MSVC 2017 15.7, XCode 10.1, and CMake 3.10. This is a setup which should be supported on Ubuntu 18.04 repositories. Minimum Boost version will also be updated though Boost is becoming less critical for the HELICS core so may not be that important. The likely minimum tested target will likely be 1.65.1 though the core might work with older versions and its use can be disabled completely. Certain features may require a newer boost version(1.70) than what would be available on Ubuntu 18.04. General target requirements will allow HELICS to build on the most recent 2 LTS versions of Ubuntu using readily available repo packages. Minimum required compilers for building on macOS and systems using ICC will include Xcode 10 and ICC 19. The minimum ZMQ version will also be bumped up to 4.2. General policy for Mac builds will be supporting Xcode compilers on versions of MacOS that receive security upgrades which is generally the last 3 versions, though 10.1+ and 11 will likely be the only 2 supported at HELICS 3.0 release, and support minor releases for at least 2 years. MSVC compilers will be supported for at least 2 years from release date, an appropriate CMake (meaning slightly newer than the compiler) will also be required for Visual Studio builds.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -8,16 +8,34 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
 
 ### Fixed
 
 - Sporadic failures in the Webserver and websocket tests
+- A bug in the translation of vectors to complex vectors
+- A bug in the copying of vectors into the C shared library
+- Numerous clang-tidy identified issues mostly for code readability
+- Some issues with the exists query not working in certain circumstances and for cores and brokers
+- The nonlings test would fail if the branch name had `error` in it. A check was put into eliminate this false negative test failure.
+- a few sporadic failure cases in the http and websocket tests
```
Comment (nightlark) [inline CHANGELOG.md:36]:
```suggestion
- A few sporadic failure cases in the http and websocket tests
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -26,10 +44,18 @@ Everything within a major version number should be code compatible (with the exc
 - "version" and "version_all" queries to get the local version string and the version strings of all the cores/brokers in the federation
 - A few missing operations to the C++98 interface for Message objects, add `helicsMessageClone` and `helicsEndpointCreateMessage` functions in the C interface. Add a test case for some of the C++98 message operations.
 - `helicsQuerySetTarget` and `helicsQuerySetQueryString` operations to modify an existing query in the C interface
-
-### Deprecated
-
-### Removed
+- a set of reduction operations for multi-input handling on inputs, options for setting input source priority and the number of expected connections.
```
Comment (nightlark) [inline CHANGELOG.md:47]:
```suggestion
- A set of reduction operations for multi-input handling on inputs, options for setting input source priority and the number of expected connections
```

**Comment by phlptp** [comment]
Anything we are missing in the Changelog?

File: CHANGELOG.md [hunk 8]
```diff
@@ -26,10 +45,18 @@ Everything within a major version number should be code compatible (with the exc
...
+- A few more parameters that can be handled in the Webserver and support for use of uuid instead of name
+- A few missing message operators to the C++98 API, such as `data`, `append`, `setFlag`, `checkFlag`, `sendMessageZeroCopy`
+- Made the Message class return a self Reference for the setters
+- A test to run the helics-broker executable as part of the CI tests
+- A bug in the helics_broker that no longer ran correct defaults
+- A "version_all" query, to retrieve the version of HELICS in use for all cores/brokers, and a "version" query to retrieve the version of a specific target.
+- A series of checks for markdown, spelling, shellcheck, python formatting, cpplint, end-of-line and automated generation of PR scripts for the formatting updates.
+- Some level of automated scaling for benchmarks for small systems
+- API functions for retrieving the build flags used to generate the library
+- Some additional message interpreters to aid in debugging
```
Comment (nightlark) [inline CHANGELOG.md:59]:
```suggestion
- Some additional message interpreters to aid in debugging
- Added a test using the standalone `helics_broker` to run an example
```

**Pull request merged**